In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("housing.csv")

In [3]:
df

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
0        -122.23     37.88                41.0        880.0           129.0   
1        -122.22     37.86                21.0       7099.0          1106.0   
2        -122.24     37.85                52.0       1467.0           190.0   
3        -122.25     37.85                52.0       1274.0           235.0   
4        -122.25     37.85                52.0       1627.0           280.0   
...          ...       ...                 ...          ...             ...   
20635    -121.09     39.48                25.0       1665.0           374.0   
20636    -121.21     39.49                18.0        697.0           150.0   
20637    -121.22     39.43                17.0       2254.0           485.0   
20638    -121.32     39.43                18.0       1860.0           409.0   
20639    -121.24     39.37                16.0       2785.0           616.0   

       population  households  median_income  median_house_value  \
0           322.0       126.0         8.3252            452600.0   
1          2401.0      1138.0         8.3014            358500.0   
2           496.0       177.0         7.2574            352100.0   
3           558.0       219.0         5.6431            341300.0   
4           565.0       259.0         3.8462            342200.0   
...           ...         ...            ...                 ...   
20635       845.0       330.0         1.5603             78100.0   
20636       356.0       114.0         2.5568             77100.0   
20637      1007.0       433.0         1.7000             92300.0   
20638       741.0       349.0         1.8672             84700.0   
20639      1387.0       530.0         2.3886             89400.0   

      ocean_proximity  
0            NEAR BAY  
1            NEAR BAY  
2            NEAR BAY  
3            NEAR BAY  
4            NEAR BAY  
...               ...  
20635          INLAND  
20636          INLAND  
20637          INLAND  
20638          INLAND  
20639          INLAND  

[20640 rows x 10 columns]

# first we should know data information

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20640 entries, 0 to 20639
Data columns (total 10 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   longitude           20640 non-null  float64
 1   latitude            20640 non-null  float64
 2   housing_median_age  20640 non-null  float64
 3   total_rooms         20640 non-null  float64
 4   total_bedrooms      20433 non-null  float64
 5   population          20640 non-null  float64
 6   households          20640 non-null  float64
 7   median_income       20640 non-null  float64
 8   median_house_value  20640 non-null  float64
 9   ocean_proximity     20640 non-null  object 
dtypes: float64(9), object(1)
memory usage: 1.6+ MB


# -----------------------------------------------

# top 5 ocean proximity from biggest to smallest

In [5]:
df["ocean_proximity"].value_counts().head(5)

<1H OCEAN     9136
INLAND        6551
NEAR OCEAN    2658
NEAR BAY      2290
ISLAND           5
Name: ocean_proximity, dtype: int64

### the biggest number in ocean proximity in <1H OCEAN there is 9136
### and the smallest number in island there is 5

# ----------------------------------------------------------

# total rooms with smallest and biggest latitude

In [12]:
df.groupby("latitude").min()["total_rooms"]

latitude
32.54     938.0
32.55    2206.0
32.56     864.0
32.57     444.0
32.58     393.0
          ...  
41.84    2677.0
41.86    2344.0
41.88    1504.0
41.92    1668.0
41.95    1867.0
Name: total_rooms, Length: 862, dtype: float64

## low latitude is 32.54 has 938 rooms and biggest latitude is 41.95
## have 1867 rooms

## ----------------------------------------------------

# table between longtude and latitude

In [13]:
df[["longitude","latitude"]]

longitude  latitude
0        -122.23     37.88
1        -122.22     37.86
2        -122.24     37.85
3        -122.25     37.85
4        -122.25     37.85
...          ...       ...
20635    -121.09     39.48
20636    -121.21     39.49
20637    -121.22     39.43
20638    -121.32     39.43
20639    -121.24     39.37

[20640 rows x 2 columns]

# --------------------------------------------------------

# biggest number of population info

In [34]:
df[df["population"].max()==df["population"]]

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
15360    -117.42     33.35                14.0      25135.0          4819.0   

       population  households  median_income  median_house_value  \
15360     35682.0      4769.0         2.5729            134400.0   

      ocean_proximity  
15360       <1H OCEAN

# maximum bedrooms

In [35]:
df["total_bedrooms"].max()

6445.0

# ------------------------------------------------------

# minimum rooms

In [36]:
df["total_rooms"].min()

2.0

# -----------------------------------------------------------

# biggest households in which ocean_proximity

In [41]:
df[df["households"].max()==df["households"]]["ocean_proximity"]

9880    <1H OCEAN
Name: ocean_proximity, dtype: object

# -----------------------------------------------------

# we want to know ocean_proximity which has 1470 population

In [43]:
df[df["population"]==1470 ]["ocean_proximity"]

518       NEAR BAY
1461        INLAND
1467        INLAND
7148     <1H OCEAN
11877       INLAND
17879    <1H OCEAN
Name: ocean_proximity, dtype: object